include/fluent-bit/flb_log.h

```diff
@@ -63,7 +63,11 @@ struct flb_log {
 
 static inline int flb_log_check(int l) {
     struct flb_worker *w;
-    w = (struct flb_worker *) FLB_TLS_GET(flb_worker_ctx);
+    if (flb_worker_ctx) {
```

Comment:
this made a lot of tests fail on osx. Let me know if you'd like me to revert it.

Reply:
I prefer always using FLB_TLS_GET(), not that systems that do not support __thread need pthread_getspecific, that macro workaround that.

Reply:
Tests don't call FLB_TLS_INIT and for some reason `pthread_getspecific(0)` returns....something on osx. Either way, it makes any log statement crash with a SIGSEGV.

Reply:
Actually - I just checked. OSX does support `__thread` - the check in CMakeLists is just checking an unrelated function as well. I'll fix that instead and revert this.